compute_tools/src/compute.rs

```diff
@@ -912,37 +912,70 @@ LIMIT 100",
                         .clone();
                 }
 
+                let ext_path = &self
+                    .ext_remote_paths
+                    .get()
+                    .expect("error accessing ext_remote_paths")[&real_ext_name];
+                let ext_archive_name = ext_path.object_name().expect("bad path");
+
+                let mut first_try = false;
+                if !self
+                    .ext_download_progress
+                    .read()
+                    .expect("lock err")
+                    .contains_key(ext_archive_name)
                 {
-                    // TODO:
-                    // started_to_download_extensions maybe should store archive paths rather than extension names...
-                    // because right now if you try to download the same archive multiple times
-                    // you will get the same archive downloaded multiple times
-                    // but, this is not such an easy fix.
-                    // because now requests should maybe pause and wait if someone else started making the download
-                    let mut started_to_download_extensions = self
-                        .started_to_download_extensions
-                        .lock()
-                        .expect("bad lock");
-                    if started_to_download_extensions.contains(&real_ext_name) {
-                        info!(
-                            "extension {:?} already exists, skipping download",
-                            &ext_name
-                        );
-                        return Ok(0);
-                    } else {
-                        started_to_download_extensions.insert(real_ext_name.clone());
+                    self.ext_download_progress
+                        .write()
+                        .expect("lock err")
+                        .insert(ext_archive_name.to_string(), (Utc::now(), false));
+                    first_try = true;
+                }
+                let (download_start, download_completed) =
+                    self.ext_download_progress.read().expect("lock err")[ext_archive_name];
+                let start_time_delta = Utc::now()
+                    .signed_duration_since(download_start)
+                    .to_std()
+                    .unwrap()
+                    .as_millis() as u64;
+
+                if download_completed {
+                    info!("extension already downloaded, skipping re-download");
+                    return Ok(0);
+                } else if start_time_delta < 3000 && !first_try {
```

Comment:
I suggest to turn this `3000` into a const

Reply:
done